drivers/memc/memc_mcux_flexspi.c

```diff
@@ -174,8 +181,16 @@ static int memc_flexspi_init(const struct device *dev)
 #define MEMC_FLEXSPI_CFG_XIP(node_id) false
 #endif
 
+#ifdef CONFIG_PINCTRL
+#define PINCTRL_DEFINE(n) PINCTRL_DT_INST_DEFINE(n);
+#define PINCTRL_INIT(n) .pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),
+#else
+#define PINCTRL_DEFINE(n)
+#define PINCTRL_INIT(n)
+#endif
+
 #define MEMC_FLEXSPI(n)							\
-	PINCTRL_DT_INST_DEFINE(n);					\
+	PINCTRL_DEFINE(n)						\
```

Comment:
```suggestion
	PINCTRL_DEFINE(n);						\
```

Reply:
Unless you feel strongly about this change, I'd prefer to keep this as is, both because this is the pattern used in the remainder of the NXP drivers that conditionally support pin control, and because this PR is fixing an issue that is broken upstream, and I'd prefer not to restart CI.

Reply:
> because this is the pattern used in the remainder of the NXP drivers

Including a semicolon at the end of macro definitions is generally a bad practice, but if there are others doing the same in the repo, I guess we can batch-fix them later.